lib/pbench/test/unit/server/test_requests.py

```diff
@@ -125,10 +125,14 @@ def test_missing_authorization_header(self, client, caplog, server_config):
         assert response.status_code == HTTPStatus.UNAUTHORIZED
         self.verify_logs(caplog)
 
-    def test_malformed_authorization_header(self, client, caplog, server_config):
+    @pytest.mark.parametrize(
+        "malformed_token", ("malformed", "bear token" "Bearer malformed"),
+    )
+    def test_malformed_authorization_header(
+        self, client, caplog, server_config, malformed_token
+    ):
```

Comment:
This appears to be intended to override `Commons.test_malformed_authorization_header()`, presumably to prevent the base class test code from running...is that correct?  (It would be nice to have the signature match as much as possible, so `caplog` should be at the end.)

So, I'm wondering how much we are benefiting from having a base class implementation (especially since we have to repeat the parameterization in the derived classes).

I'm thinking that maybe a different approach might be better.  For instance, we could have a base class test shell (i.e., named `test_malformed_authorization_header()`), which the subclasses inherit and run, and it could invoke a class helper method which the subclasses can override.  In this way, all instances of the test use the same parameterization, and subclasses like this one here could replace the request if they want to; the only question in my mind is whether fixtures can be used in helper methods, or whether that magic only works on "test" methods.

Reply:
> So, I'm wondering how much we are benefiting from having a base class implementation (especially since we have to repeat the parameterization in the derived classes).

This overriding is happening only for the `Dataset_publish` because for Dataset_publish we need an additional `attach_dataset` fixture. 

Right now the flow of logic we have in our dataset publish is that we query the dataset first and then the user (Since we do not require user parameter from the client) and then we check the authorization. A user can only be queried from the dataset as the owner. And thats why I have to override the malformed authorization test in publish since we need a dataset before we check the authentication. 


> I'm thinking that maybe a different approach might be better. For instance, we could have a base class test shell (i.e., named `test_malformed_authorization_header()`), which the subclasses inherit and run, and it could invoke a class helper method which the subclasses can override. In this way, all instances of the test use the same parameterization, and subclasses like this one here could replace the request if they want to; the only question in my mind is whether fixtures can be used in helper methods, or whether that magic only works on "test" methods.

If the fixtures are defined on helper function and if we are calling the helper function then we have to explicitly provide value of the fixture to that helper function just like what we are doing with this test function. Function defined with `Pytest` decorator gets called only when the pytest invokes it and pytest only invoke function with `test` prefix. If we were to call this function from another function the pytest fixture wont automatically  get called and so we have to explicitly provide that value.

Reply:
Thank you for setting me straight:  this class is _not_ derived from `Commons`, and so this is non-issue.